spec/scoping/filter_spec.rb

```diff
@@ -0,0 +1,31 @@
+require "spec_helper"
+
+RSpec.describe Graphiti::Scoping::Filter do
```

Comment:
I typically rely more on integration-y specs instead of mocking the order of calls. If we moved this spec to `spec/filtering_spec.rb`, what value would we lose by removing this file?

Reply:
@richmolj  I get your point; definitely having integration-y specs makes it easy to maintain when the underlying implementation changes. Though, for specific "core" components within the implementation might help to have unit tests - as this will monitor specific behavior of these components (the "issue" was within the implementation of this class).
Given the actual content of the spec, I would say that, right now, this can be moved as you suggested.
@marcusjanke what do you say?

Reply:
@mihaimuntenas I agree in the broad strokes, especially after we release 1.0. As it stands now, I think the tests in this PR are perfect.